src/kOS/Screen/TermWindow.cs

```diff
@@ -89,7 +89,7 @@ public class TermWindow : KOSManagedWindow , ITermWindow
         private KOSTextEditPopup popupEditor;
 
         // data stored per telnet client attached:
-        private readonly List<TelnetSingletonServer> telnets; // support exists for more than one telnet client to be attached to the same terminal, thus this is a list.
+        private volatile List<TelnetSingletonServer> telnets; // support exists for more than one telnet client to be attached to the same terminal, thus this is a list.
```

Comment:
The writing thread in the telnet does end up trying to edit this list, when it detects that the user hit ctrl-D to detach the terminal from the CPU.  (not directly, but as it tries to shut everything down about the association with the terminal, one of the methods it's calling ends up deleting the telnet from this list.)

That's why this was changed to volatile to remind C# it can be thread-accessed.